libyang/data.py

```diff
@@ -224,6 +281,31 @@ def validate(
         if ret != 0:
             raise self.context.error("validation failed")
 
+    def diff(
+        self,
+        other: "DNode",
+        no_siblings: bool = False,
+        with_defaults: bool = False,
+    ) -> Iterator[DDiff]:
+        flags = diff_flags(no_siblings=no_siblings, with_defaults=with_defaults)
+        dlist = lib.lyd_diff(self.cdata, other.cdata, flags)
+        if not dlist:
+            raise self.context.error("diff failed")
+
+        i = 0
+        try:
+            while dlist.type[i] != lib.LYD_DIFF_END:
+                first = (
+                    self.new(self.context, dlist.first[i]) if dlist.first[i] else None
+                )
+                second = (
+                    self.new(self.context, dlist.second[i]) if dlist.second[i] else None
+                )
```

Comment:
Indeed, it is not pretty...

Maybe use a verbose `if`...`else` instead?

Reply:
Feel free to change.

Reply:
I really hate this formatter.

Reply:
I changed it again. looks better now.